File: android/app/src/main/java/com/schoolfinder/app/di/ServiceLocator.java
package com.schoolfinder.app.di;

import android.content.Context;

import com.schoolfinder.app.BuildConfig;
import com.schoolfinder.app.data.SchoolRepository;
import com.schoolfinder.app.data.SessionStore;
import com.schoolfinder.app.data.remote.Network;

/** Tiny manual DI container — avoids pulling in Hilt for a small app. */
public final class ServiceLocator {

    private ServiceLocator() {}

    private static SchoolRepository repository;
    private static SessionStore sessionStore;

    public static void init(Context context) {
        if (repository != null) return;
        repository = new SchoolRepository(Network.create(BuildConfig.API_BASE_URL));
        sessionStore = new SessionStore(context.getApplicationContext());
    }

    public static SchoolRepository repository() {
        return repository;
    }

    public static SessionStore sessionStore() {
        return sessionStore;
    }
}

File: android/app/src/main/java/com/schoolfinder/app/di/ServiceLocator.kt
package com.schoolfinder.app.di

import android.content.Context
import com.schoolfinder.app.BuildConfig
import com.schoolfinder.app.data.SchoolRepository
import com.schoolfinder.app.data.SessionStore
import com.schoolfinder.app.data.remote.Network

/** Tiny manual DI container — avoids pulling in Hilt for a small app. */
object ServiceLocator {

    lateinit var repository: SchoolRepository
        private set

    lateinit var sessionStore: SessionStore
        private set

    fun init(context: Context) {
        if (::repository.isInitialized) return
        val api = Network.create(BuildConfig.API_BASE_URL)
        repository = SchoolRepository(api)
        sessionStore = SessionStore(context.applicationContext)
    }
}
